src/Internal/Types/Interval.purs

```diff
@@ -232,23 +238,24 @@ derive instance Functor UpperBound
 instance Show a => Show (UpperBound a) where
   show = genericShow
 
--- | An interval of `a`s.
+-- | An interval of `a`s in the same sense as in plutus.
 -- |
 -- | The interval may be either closed or open at either end, meaning
 -- | that the endpoints may or may not be included in the interval.
 -- |
 -- | The interval can also be unbounded on either side.
-newtype Interval :: Type -> Type
-newtype Interval a = Interval { from :: LowerBound a, to :: UpperBound a }
+newtype PlutusInterval :: Type -> Type
+newtype PlutusInterval a = PlutusInterval
+  { from :: LowerBound a, to :: UpperBound a }
 
-derive instance Generic (Interval a) _
-derive newtype instance Eq a => Eq (Interval a)
-derive instance Functor Interval
-derive instance Newtype (Interval a) _
+derive instance Generic (PlutusInterval a) _
+derive newtype instance Eq a => Eq (PlutusInterval a)
+derive instance Functor PlutusInterval
+derive instance Newtype (PlutusInterval a) _
 
 instance
-  HasPlutusSchema (Interval a)
-    ( "Interval"
+  HasPlutusSchema (PlutusInterval a)
+    ( "PlutusInterval"
```

Comment:
This is the more concerning change to me, I don't really know if this is fine to do.

Reply:
No. What was the motivation for renaming?

Reply:
Why do we ever need it?

Reply:
If we use the name `Interval` for the new interval style and users have been using `to`, `from`, `never`, `always` and `mkInterval` instead of the old `Interval` constructor, then they only need to replace `mkInterval` with `mkFiniteInterval` to use the new type, that's why I renamed the old type to a new name `PlutusInterval`. 

Original I keep `PlutusInterval` since I don't properly understand `HasPlutusSchema`, `ToData`, and `FromData`, so It was easy to just translate a new `Interval` to `PlutusInterval` and use its instances. After some thought, `PlutusInterval` is interesting to keep as it follows the `Plutus` code and it can help to understand (and debug) the new `Interval` type. 

Of course, we still need to rewrite the instances of `HasPlutusSchema`, `ToData`, and `FromData` for the new `Interval` type to avoid using `PlutusInterval`